crates/spell-event-bus/src/bus.rs

```diff
@@ -0,0 +1,299 @@
+        let this = Self {
+            sources,
+            recv_cmd_channel,
+            send_events,
+        };
+        (this, api, recv_events)
+    }
+
+    pub fn start(self) -> task::JoinHandle<()> {
+        task::spawn(self.run())
+    }
+
+    async fn run(self) {
+        let send_events = self.send_events;
+
+        let mut recv_cmd_channel = self.recv_cmd_channel.fuse();
+        let sources = self
+            .sources
+            .into_iter()
+            .map(|source| source.fuse())
+            .collect::<Vec<_>>();
+        let mut sources_channel = stream::select_all(sources);
+
+        let mut state = SubscribersState::new();
+        loop {
+            let now = Instant::now();
+            // Wait until the next spell should be awaken. If there are no spells wait for unreachable amount of time,
+            // which means that timer won't be triggered at all. We overwrite the timer each loop (aka after each event)
+            // to ensure that we don't miss newly scheduled spells.
+            let mut timer = {
+                let next_scheduled_in = state.next_scheduled_in(now).unwrap_or(Duration::MAX);
+                async_std::stream::interval(next_scheduled_in).fuse()
+            };
+
+            let result: Result<(), BusError> = try {
+                select! {
+                    command = recv_cmd_channel.select_next_some() => {
+                        match command {
+                            Command::Subscribe(spell_id, config) => {
+                                state.subscribe(spell_id.clone(), config)?;
+                            },
+                            Command::Unsubscribe(spell_id, reply) => {
+                                state.unsubscribe(&spell_id);
+                                reply.send(()).map_err(|_| BusError::Unsubscribe(spell_id))?;
+                            },
+                        }
+                    },
+                    event = sources_channel.select_next_some() => {
+                        for spell_id in state.subscribers(&event.get_type()) {
+                            let event = Event::Peer(event.clone());
+                            Self::notify_spell(&send_events, spell_id, event)?;
+                        }
+                    },
+                    _ = timer.select_next_some() => {
+                        // The timer is triggered only if there are some spells to be awaken.
+                        let scheduled_spell = state.scheduled.pop().expect("billions of years have gone by already?");
+                        Self::notify_spell(&send_events, &scheduled_spell.data.id, Event::Timer)?;
+                        let rescheduled = Scheduled::at(scheduled_spell.data, Instant::now())?;
+                        state.scheduled.push(rescheduled);
+                    },
+                }
+            };
+            if let Err(e) = result {
+                log::error!("{}", e);
+            }
+        }
+    }
+
+    fn notify_spell(
```

Comment:
Event Bus _notifies_ spells about _events_ that happened according to _trigger_ config. 3 different words for the same thing.

How about this?
```suggestion
    fn trigger_spell(
```